src/descriptor.rs

```diff
@@ -148,10 +152,145 @@ impl Descriptor {
     }
 
     /// Constructs a new [`Descriptor`] from a byte [`Vec`].
+    ///
+    /// # Warning
+    ///
+    /// Users are advised to use the `new_*` methods whenever possible.
     pub fn from_vec(bytes: Vec<u8>) -> Result<Self, DescriptorError> {
         Self::from_bytes(&bytes)
     }
 
+    /// Constructs a new [`Descriptor`] from an `OP_RETURN` payload.
+    ///
+    /// The payload is expected to be less than 80 bytes.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bitcoin_bosd::{Descriptor, DescriptorType};
+    /// let payload = b"hello world";
+    /// let desc = Descriptor::new_op_return(payload).expect("valid payload under 80 bytes");
+    /// # assert_eq!(desc.type_tag(), DescriptorType::OpReturn);
+    /// # assert_eq!(desc.payload(), b"hello world");
+    /// ```
+    pub fn new_op_return(payload: &[u8]) -> Result<Self, DescriptorError> {
+        let type_tag = DescriptorType::OpReturn;
+        let payload_len = payload.len();
+        if payload_len > MAX_OP_RETURN_LEN {
+            Err(DescriptorError::InvalidPayloadLength(payload_len))
+        } else {
+            Ok(Self {
+                type_tag,
+                payload: payload.to_vec(),
+            })
+        }
+    }
+
+    /// Constructs a new [`Descriptor`] from a P2PKH payload.
+    ///
+    /// The payload is expected to be a valid 20-byte hash.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bitcoin_bosd::{Descriptor, DescriptorType};
+    /// let payload = [0u8; 20]; // all zeros, don't use in production
+    /// let desc = Descriptor::new_p2pkh(&payload);
+    /// # assert_eq!(desc.type_tag(), DescriptorType::P2pkh);
+    /// # assert_eq!(desc.payload(), [0u8; 20]);
+    /// ```
+    pub fn new_p2pkh(payload: &[u8; 20]) -> Self {
+        let type_tag = DescriptorType::P2pkh;
+        Self {
+            type_tag,
+            payload: payload.to_vec(),
+        }
+    }
+
+    /// Constructs a new [`Descriptor`] from a P2SH payload.
+    ///
+    /// The payload is expected to be a valid 20-byte hash.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bitcoin_bosd::{Descriptor, DescriptorType};
+    /// let payload = [0u8; 20]; // all zeros, don't use in production
+    /// let desc = Descriptor::new_p2sh(&payload);
+    /// # assert_eq!(desc.type_tag(), DescriptorType::P2sh);
+    /// # assert_eq!(desc.payload(), [0u8; 20]);
+    /// ```
+    pub fn new_p2sh(payload: &[u8; 20]) -> Self {
```

Comment:
Use the corresponding constant.

Reply:
Yes, in retrospect obvious :/

Will propagate the directive to all of them.

Reply:
Done in 99923f646f9b5e6bb68b5f6ad47b2f8c17f3ba5c